File: scripts/approved-site-wording.mjs
import fs from 'node:fs';
const changes=JSON.parse(fs.readFileSync(new URL('./ltspice-site-wording.json',import.meta.url),'utf8'));
const plain=s=>s.replace(/\[([^\]]+)\]\([^)]*\)/g,'$1').replace(/[*`]/g,'').replace(/^#+ /,'').replace(/ \{#[^}]+\}/g,'');
export function restoreSiteWording(file,text,rendered=false){
  if(rendered){
    if(file.endsWith('/PLL.md')) text=text.replaceAll('first-order model','first-order teaching model');
    text=text.replaceAll('Logic presets','Teaching presets');
  }
  for(const record of changes.filter(c=>c.path===file)) for(const change of record.changes){
    const after=rendered?plain(change.after):change.after;
    const before=rendered?plain(change.before):change.before;
    text=text.replace(after,before);
  }
  return text;
}

File: scripts/approved_site_wording.py
import json
from pathlib import Path
changes=json.loads(Path(__file__).with_name('ltspice-site-wording.json').read_text(encoding='utf-8'))
placements=json.loads(Path(__file__).with_name('ltspice-placement.json').read_text(encoding='utf-8'))
def restore_placement(file,text):
 for record in placements:
  if record['path']==file:
   for block in reversed(record['inserts']):
    assert text.count(block)==1,file
    text=text.replace(block,'\n')
   for change in record['replacements']:
    assert text.count(change['after'])==1,file
    text=text.replace(change['after'],change['before'])
 return text
def restore_site_wording(file,text):
 text=restore_placement(file,text)
 for record in changes:
  if record['path']==file:
   for change in record['changes']:
    assert text.count(change['after'])==1,(file,change['after'])
    text=text.replace(change['after'],change['before'])
 return text
